### PR TITLE
fix goroutine leak in webhook filter

### PR DESCRIPTION
**Pull request by szuecs**
fix goroutine leak in webhook filter that you can observe if dataclients trigger regular updates and this filter will be recreated. A common fix is the way tokeninfo caches the authclient by target

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

### CODE REVIEW

File: filters/auth/webhook.go [hunk 1]
```diff
@@ -89,14 +90,21 @@ func (ws *webhookSpec) CreateFilter(args []interface{}) (filters.Filter, error)
 		}
 	}
 
-	ac, err := newAuthClient(s, webhookSpanName, ws.options.Timeout, ws.options.MaxIdleConns, ws.options.Tracer)
-	if err != nil {
-		return nil, filters.ErrInvalidFilterParameters
+	var ac *authClient
+	var err error
+	if ac, ok = webhookAuthClient[s]; !ok {
```
[the remark this answers is not in the document]
Comment (AlexanderYastrebov) [reply filters/auth/webhook.go:95]: Hm, I think we would need to remove Close method for all filters that cache clients, so lets do it separately

**Comment by AlexanderYastrebov** [comment]
:+1:

File: filters/auth/webhook.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (szuecs) [reply filters/auth/webhook.go:95]: Good catch!

**Comment by szuecs** [comment]
:+1:

**Pull request merged**